### PR TITLE
Make getDerivedHDAddress return address, not key

### PR DESCRIPTION
**Pull request by chromatic**
Addresses GH #217.

I proved the first test correct with a third-party derivation library; looking for feedback before I go further.

### CODE REVIEW

**Comment by xanimo** [comment]
So I think we should leave the old one unchanged (due to not wanting breaking changes that other API's depend on) and duplicate that function where you made those changes and add a new suffix, e.g. getDerivedHDAddressAsP2PKH. I think the reason for this confusion is lack of specificity re naming conventions, and referencing from sources like [Learn me a bitcoin](https://learnmeabitcoin.com/technical/keys/hd-wallets/extended-keys/#address) wherein the dgub 112 char key is mentioned as an address and not a p2pkh legacy address.

image

**Comment by chromatic** [comment]
That makes sense and I'm content to rework (or let someone else rework) this PR for that approach. If that's the approach you prefer, I do think clarifying the documentation of both functions is valuable.

**Comment by xanimo** [comment]
Yeah if you wouldn't mind reworking those functions while leaving getDerivedHDAddress the same, it would be much appreciated.

**Comment by chromatic** [comment]
I've rewritten my commit to go in that direction. This has tests but not documentation in `docs/` or examples.

**Comment by xanimo** [comment]
Looks good. Will do a formal review later this afternoon.

File: src/address.c [hunk 1]
```diff
@@ -490,6 +490,36 @@ int getDerivedHDAddress(const char* masterkey, uint32_t account, bool ischange,
...
+/**
+ * @brief This function generates a derived child address from a masterkey using
+ * a BIP44 standardized static, non hardened path comprised of an account, a change or
+ * receiving address and an address index.
+ *
+ * @param masterkey The master key from which children are derived from.
+ * @param account The account that the derived address would belong to.
+ * @param ischange Boolean value representing either a change or receiving address.
+ * @param addressindex The index of the receiving/change address per account.
+ * @param outp2pkh The derived address in P2PSH form.
```
Comment (xanimo) [inline src/address.c:502]: nit: P2PKH

File: test/address_tests.c [hunk 2]
```diff
@@ -134,6 +134,32 @@ void test_address()
...
     u_assert_str_eq(extout, "dgub8wfrZMXz8ojFcPziSubEoQ65sB4PYPyYTMo3PqFwf2Vx5zZ6ia17Nk2Py25c3dvq1e7ZnfBrurCS5wuagzRoBCXhJ2NeGU54NBytvuUuRyA");
 
+    /* ckd p2pkh generation */
+    res = getDerivedHDAddressAsP2PKH(masterkey_main_ext, 0, false, 0, extout);
+    u_assert_int_eq(res, true);
+    u_assert_str_eq(extout, "DCm7oSg95sxwn3sWxYUDHgKKbB2mDmuR3B");
+    res = getDerivedHDAddressAsP2PKH(masterkey_main_ext, 0, true, 0, extout);
+    u_assert_int_eq(res, true);
+    u_assert_str_eq(extout, "D91jVi3CVGhRmyt83fhMdL4UJWtDuiTZET");
+    res = getDerivedHDAddressAsP2PKH(masterkey_main_ext, 0, false, 0, extout);
```
Comment (xanimo) [inline test/address_tests.c:144]: nit: L144-148 and L156-161 are duplicate test cases

**Review by xanimo — COMMENTED** [review]
Requesting fixes for a couple nits.

**Comment by edtubbs** [comment]
This looks good. I agree with the other comments and on updating `address.md` and `example.c` as you mentioned. Thanks.

**Review by michilumin — APPROVED** [review]
Good change, logical - thanks for that input @chromatic . Appreciated.

**Pull request merged**